snippets/olm-psa-opm-catalog-source-restricted-known-issue.adoc

```diff
@@ -0,0 +1,18 @@
+// Text snippet included in the following assemblies:
+//
+// * 
+//
+// Text snippet included in the following modules:
+//
+// * olm-catalogsource.adoc
+// * olm-creating-catalog-from-index
+
+:_content-type: SNIPPET
+
+There is a known issue in the `opm` CLI tool that prevents SQLite-based catalog source pods from reaching a ready state under restricted pod security enforcement. (link:https://issues.redhat.com/browse/OCPBUGS-122[*OCPBUGS-122*])
+
+Pod security admissions were introduced in Kubernetes 1.24 to ensure pod security standards. In {product-title} 4.11 and 4.12, namespaces do not have restricted pod security enforcement by default. Default restricted enforcement for namespaces is planned for a future release of {product-title}. When restricted enforcement occurs, the security context of the pod specification for catalog source pods must match the restricted pod security standard. If your catalog source image requires a different pod security standard, the pod security admissions label for the namespace must be explicitly set.
+
+The `openshift-marketplace` namespace is explicitly labeled as privileged. Catalog sources in that namespace are not affected by this known issue.
+
+A workaround is available for cluster administrators. Catalog authors can prevent the known issue by updating the `opm` CLI tool or migrating their catalog to the file-based catalog format. For more information about the issue and how to set the pod security level for namespaces, see link:https://access.redhat.com/articles/6977554[SQLite-based CatalogSource pod does not become ready due to "Error: open ./db-<number>: permission denied"] (Red Hat Knowledgebase).
```

Comment:
that's not a workaround, that's a solution. (upgrade the catalog image to a version of opm that can run restricted).

the workaround (for if you want to keep using the existing catalog image) is to:
1) explicitly set the security_mode to legacy (not strictly needed since it's currently the default, but a good idea to ensure you don't get broken in the future if/when the default changes)
2) label the NS the catalog is for PSA enforcement baseline or privileged (to avoid issues when the default enforcement changes to restricted in the future)

Note that in 4.12 you don't need to do either of those things since:
1) the default security_mode is legacy
2) the default NS PSA enforcement is privileged

but both of those defaults are subject to change in the future, at which point you'll be impacted if you don't take action now.